Release 1.4.0 into develop

AliSoftware opened the pull request:
New version 1.4.0

AliSoftware committed:
Bumped to version 1.4.0

mokagio commented:
Just a note here on the failing GitHub Actions build because:

```
Unable to find a spec satisfying nokogiri (~> 1.11) in the set. Perhaps the
lockfile is corrupted? Found nokogiri (1.11.7-x86_64-darwin) that did not match
the current platform.
```

I need to learn more about how Bundler 2.2.x manages the platforms, but I think we can _work around_ this by adding the following to `.bundle/config`:

```
BUNDLE_SPECIFIC_PLATFORM: "false"
```

(_We could also ignore this and move on, but isn't it unsatisfying?_)

mokagio committed:
Disable platform specific gem installs via Bundler

See discussion here: https://github.com/wordpress-mobile/release-toolkit/pull/292#issuecomment-891758279

To ensure the setting is persisted for every user of the repository, I
decided to track the Bundler config file. This seems like a good idea
anyway, even though Bundler currently has a bug resulting in that file
being changed sometimes, which can be confusing for developers.

See https://github.com/rubygems/rubygems/issues/3708

mokagio committed:
Skip test gems group install in GitHub Actions workflow

We don't need them because we don't run the tests in that workflow.

mokagio committed:
Remove no-op option `--with screenshots` from CircleCI `bundle install`

There is no "screenshots" group. That option didn't produce any effect.

mokagio committed:
Add CHANGELOG entry for #293

mokagio commented:
Opened #293 to address the CI issue.

mokagio reviewed (approved):
LGTM. 

Once #293 gets into the release branch, we should be able to merge and ship.

AliSoftware committed:
Merge pull request #293 from wordpress-mobile/disable-platform-specific-gems

AliSoftware commented on CHANGELOG.md:
```diff
@@ -29,6 +39,9 @@ _None_
 
+### Internal Changes
+
+* Opt-out from installing platform-specific gems with Bundler [#293]
```
Merged 293 too quick and didn't realize this was in the wrong place, needs to be under 1.4.0 not 1.3.1

AliSoftware committed:
Move 293 entry of CHANGELOG in right section

AliSoftware committed:
bundle update

AliSoftware committed:
Add CHANGELOG entry

AliSoftware committed:
Merge pull request #294 from wordpress-mobile/bundle-update

Fix security issues with old gems
